python/runtime/cudaq/platform/py_alt_launch_kernel.cpp

```diff
@@ -71,8 +71,6 @@ jitAndCreateArgs(const std::string &name, MlirModule module,
                  std::size_t startingArgIdx = 0) {
   ScopedTraceWithContext(cudaq::TIMING_JIT, "jitAndCreateArgs", name);
   auto mod = unwrap(module);
-  auto cloned = mod.clone();
-  auto context = cloned.getContext();
```

Comment:
@schweitzpgi - `cloned` is a `ModuleOp` that was presumably not entirely cleaned up when going out of scope, even when it was never used beyond these two lines in this function. Moving the `mod.clone()` into the rarely-entered `if` statement below made most of one of the memory leaks go away.

Reply:
This is curious. I wonder why it isn't cleaned up properly in one case but is, apparently, cleanup up properly if placed in a different scope. One would expect the dtor to do the same thing either way.

Reply:
It may *also* not be cleaned up in the other scope, but that code path happens far less often, so if there is a leak, this changes makes the leak happen less often.